keystorage/aws/src/main/java/tech/pegasys/signers/aws/AwsSecretsManager.java

```diff
@@ -68,6 +80,39 @@ public Optional<String> fetchSecret(final String secretName) {
     }
   }
 
+  private Collection<SecretListEntry> listSecrets(final AbstractMap<String, String> tags) {
+    final ListSecretsRequest.Builder listSecretsRequestBuilder = ListSecretsRequest.builder();
+    if (tags != null) {
+      final Filter keys = Filter.builder().key("tag-key").values(tags.keySet()).build();
+      final Filter values = Filter.builder().key("tag-value").values(tags.values()).build();
+      listSecretsRequestBuilder.filters(keys, values);
+    }
+    final ListSecretsRequest listSecretsRequest = listSecretsRequestBuilder.build();
```

Comment:
Should we also set the maxResults as well? Not sure what the default is but don't think it is the maximum of 100.